scripts/west_commands/runners/qemu.py

```diff
@@ -4,28 +4,210 @@
+
+            flash_file = self.elf_name
+
+            if self.file_type != FileType.ELF:
+                err = 'Cannot flash; qmue runner only supports elf files'
+                raise ValueError(err)
+
+        self.get_qemu_opts()
+
+        if self.elf_name is not None and os.path.isfile(self.elf_name):
+            cmd = ([self.commander] +
+                   (['-nographic']) +
+                   ['-machine', self.machine] +
+                   (['-cpu', f'{self.cpu}'] if self.cpu else []) +
+                   (['-dtb', f'{self.dtb}'] if self.dtb else []) +
+                   (['-serial', f'pipe:{self.pipe}'] if self.pipe else []) +
+                   (['-kernel', flash_file] if not self.use_loader else []) +
+                   (self.qemu_opts_content if self.qemu_opts_content else []) +
+                   self.tool_opt)
+            kwargs = {}
+            #if not self.logger.isEnabledFor(logging.DEBUG):
+            #    kwargs['stdout'] = subprocess.DEVNULL
+            self.logger.info("flashing with {}".format(" ".join(cmd)))
+            self.logger.info("press ctrl+a x to quit qemu")
+            self.check_call(cmd, **kwargs)
+        else:
+            err = 'Cannot flash; qmue runner only supports elf files'
+            raise ValueError(err)
+
+
+    def debug(self, **kwargs):
+        flash_file = self.elf_name
+
+        if self.file is not None:
+            # use file provided by the user
+            if not os.path.isfile(self.file):
+                err = 'Cannot flash; file ({}) not found'
+                raise ValueError(err.format(self.file))
+
+            flash_file = self.elf_name
+
+            if self.file_type != FileType.ELF:
+                err = 'Cannot flash; qmue runner only supports elf files'
+                raise ValueError(err)
+
+        self.get_qemu_opts()
+
+        if self.elf_name is not None and os.path.isfile(self.elf_name):
+            server_cmd = ([self.commander] +
+                   (['-nographic']) +
+                   ['-machine', self.machine] +
+                   (['-cpu', f'{self.cpu}'] if self.cpu else []) +
+                   (['-dtb', f'{self.dtb}'] if self.dtb else []) +
+                   (['-kernel', flash_file] if not self.use_loader else []) +
+                   (self.qemu_opts_content if self.qemu_opts_content else []) +
+                   (['-s']) + (['-S']) +
+                   self.tool_opt)
+            client_cmd = (self.gdb_cmd +
+                          self.tui_arg +
+                          [self.elf_name] +
+                          ['-ex', 'target remote {}:{}'.format(self.gdb_host, self.gdb_port)] +
+                          ['-ex', 'monitor halt',
+                           '-ex', 'monitor reset',
+                           '-ex', 'load'])
+            self.run_server_and_client(server_cmd, client_cmd)
+        else:
+            err = 'Cannot flash; qmue runner only supports elf files'
+            raise ValueError(err)
```

Comment:
Same here; isn't the error handling already done above?

Reply:
it is to check the file existing, if the file is not existing, this will cause unwanted exception.